integration-test/android.Tests.ps1

```diff
@@ -0,0 +1,161 @@
+        Remove-Item -Path "$PSScriptRoot/mobile-app" -Recurse -Force -ErrorAction SilentlyContinue
+        Copy-Item -Path "$PSScriptRoot/net9-maui" -Destination "$PSScriptRoot/mobile-app" -Recurse -Force
+        Push-Location $PSScriptRoot/mobile-app
+
+        function InstallAndroidApp
+        {
+            param([string] $Dsn)
+            $dsn = $Dsn.Replace('http://', 'http://key@') + '/0'
+
+            # replace {{SENTRY_DSN}} in MauiProgram.cs
+            (Get-Content MauiProgram.cs) `
+                -replace '\{\{SENTRY_DSN\}\}', $dsn `
+            | Set-Content MauiProgram.cs
+
+            $arch = [System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture.ToString().ToLower()
+            $rid = "android-$arch"
+
+            Write-Host "::group::Build Sentry.Maui.Device.IntegrationTestApp.csproj"
+            dotnet build Sentry.Maui.Device.IntegrationTestApp.csproj `
+                --configuration Release `
+                --framework $tfm `
+                --runtime $rid
+            | ForEach-Object { Write-Host $_ }
+            Write-Host '::endgroup::'
+            $LASTEXITCODE | Should -Be 0
+
+            Write-Host "::group::Install bin/Release/$tfm/$rid/io.sentry.dotnet.maui.device.integrationtestapp-Signed.apk"
+            xharness android install -v `
+                --app bin/Release/$tfm/$rid/io.sentry.dotnet.maui.device.integrationtestapp-Signed.apk `
+                --package-name io.sentry.dotnet.maui.device.integrationtestapp `
+                --output-directory=test_output
+            | ForEach-Object { Write-Host $_ }
+            Write-Host '::endgroup::'
+            $LASTEXITCODE | Should -Be 0
+        }
+
+        function RunAndroidApp
+        {
+            param(
+                [string] $Dsn,
+                [string] $TestArg = 'None'
+            )
+
+            # Setup port forwarding for accessing sentry-server at 127.0.0.1:8000 from the emulator
+            $port = $Dsn.Split(':')[2].Split('/')[0]
+            xharness android adb -v -- reverse tcp:$port tcp:$port
+
+            Write-Host "::group::Run Android app (TestArg=$TestArg)"
+            xharness android adb -v `
+                -- shell am start -S -n io.sentry.dotnet.maui.device.integrationtestapp/.MainActivity `
+                -e SENTRY_TEST_ARG $TestArg
+            | ForEach-Object { Write-Host $_ }
+            Write-Host '::endgroup::'
+            $LASTEXITCODE | Should -Be 0
+
+            do
+            {
+                Write-Host "Waiting for app..."
+                Start-Sleep -Seconds 1
+
+                $procid = (& xharness android adb -- shell pidof "io.sentry.dotnet.maui.device.integrationtestapp") -replace '\s', ''
+                $activity = (& xharness android adb -- shell dumpsys activity activities) -match "io\.sentry\.dotnet\.maui\.device\.integrationtestapp"
+
+            } while ($procid -and $activity)
+
+            xharness android adb -v -- reverse --remove tcp:$port
```

Comment:
Should we put this in a try/finally block?

```
try {
    # Setup port forwarding
    xharness android adb -v -- reverse tcp:$port tcp:$port
    # ... run test logic 
} finally {
    # Always remove port forwarding
    xharness android adb -v -- reverse --remove tcp:$port
}
```

Reply:
Good idea. Done in https://github.com/getsentry/sentry-dotnet/pull/4559/commits/a8c25a7369c43a73a9264dc6c063302cc6c5fab7